src/Groups/tom.jl

```diff
@@ -0,0 +1,613 @@
+end
+
+function Base.:*(chi::GAPGroupMarksVector, psi::GAPGroupMarksVector)
+  @req parent(chi) === parent(psi) "tables of marks must be identical"
+  vchi = GapObj(chi)
+  vpsi = GapObj(psi)
+  n = min(length(vchi), length(vpsi))
+  vector = GapObj([vchi[i] * vpsi[i] for i in 1:n])
+  return marks_vector(parent(chi), vector)
+end
+
+function Base.zero(chi::GAPGroupMarksVector)
+  return marks_vector(parent(chi), GapObj([]))
+end
+
+function Base.one(chi::GAPGroupMarksVector)
+  val = ZZ(1)
+  return marks_vector(parent(chi), [val for i in 1:length(chi)])
+end
+
+function Base.:*(n::IntegerUnion, chi::GAPGroupMarksVector)
+  return marks_vector(parent(chi), n * values(chi))
+end
+
+function Base.:^(chi::GAPGroupMarksVector, n::IntegerUnion)
+  vchi = GapObj(chi)
+  return marks_vector(parent(chi), GapObj([vchi[i]^n for i in 1:length(vchi)]))
+end
+
+@doc raw"""
+    coordinates(::Type{T} = ZZRingElem, chi::GAPGroupMarksVector)
+                   where T <: Union{IntegerUnion, QQFieldElem}
+
+Return the vector $[a_1, a_2, \ldots, a_n]$ such that `chi`
+is equal to $\sum_{i=1}^n a_i t[i]$ where $t$ is `parent(chi)`.
+
+The result is an instance of `Vector{T}`.
+Note that the result can be shorter than `ncols(parent(chi))`.
+
+# Examples
+```jldoctest
+julia> tom = table_of_marks(symmetric_group(4));
+
+julia> chi = tom[3] * tom[6]
+marks_vector(table of marks of Sym(4), ZZRingElem[72, 0, 4])
+
+julia> println(coordinates(Int, chi))
+[2, 0, 2]
+```
+"""
+coordinates(chi::GAPGroupMarksVector) = coordinates(ZZRingElem, chi)
+
+function coordinates(::Type{T}, chi::GAPGroupMarksVector) where T <: Union{Integer, ZZRingElem, QQFieldElem}
+  c = GAPWrap.DecomposedFixedPointVector(GapObj(parent(chi)), GapObj(chi))
+  return Vector{T}(c)::Vector{T}
+end
+
+
+#############################################################################
+##
+##  interface between tables of marks and character tables
+##
+##  A table of marks and the corresponding character table store each other
+##  in attributes.
+##
+
+@doc raw"""
+    character_table(tom::GAPGroupTableOfMarks)
+
+Return the character table of the group of `tom`.
+If `tom` belongs to the library of tables of marks then the corresponding
+character table from the library of character tables is returned,
+otherwise the character table of `group(tom)`.
+
+# Examples
+```jldoctest
+julia> g = symmetric_group(3);  tom = table_of_marks(g);
+
+julia> character_table(tom) == character_table(g)
+true
+```
+"""
+function character_table(tom::GAPGroupTableOfMarks)
+  tbl = get_attribute!(tom, :character_table) do
+    if GAPWrap.IsLibTomRep(GapObj(tom))
+      info = GAP.Globals.LIBLIST.TOM_TBL_INFO
+      pos = findfirst(isequal(GapObj(lowercase(identifier(tom)))), Vector{GapObj}(info[1]))
+      pos !== GAP.Globals.fail && return character_table(string(info[2][pos]))
+    end
+    return character_table(group(tom))
+  end
+
+  set_attribute!(tbl, :table_of_marks, tom)
```

Comment:
Can't you move that into the do-end block? Then this method could also make use of `@attr`

Reply:
yes, this works